src/glyph/serialize.rs

```diff
@@ -33,41 +38,42 @@ impl Glyph {
         }
 
         let mut start = BytesStart::borrowed_name(b"advance");
-        if self.width != 0. {
-            start.push_attribute(("width", self.width.to_string().as_str()));
-        }
+        let mut have_advance = false;
```

Comment:
see my patch, I think it's simpler to just check both width & height first, and put everything else in a conditional block?